R/finalize_plot.R

```diff
@@ -321,19 +321,19 @@ finalize_plot <- function(plot = NULL,
   # set caption textbox alignment options
   if(caption_valign == "top"){
```

Comment:
with the improvements you've implemented to margins, specifically specifying margin_title_b and margin_caption_b separately, I think we can dramatically simplify this by setting the height of the grob to `grid::unit(consts$height - consts$margin_topline_tb - consts$margin_title_b - consts$margin_caption_b, "bigpts") - grid::grobHeight(grob_title)` and alternating only the alignment of the text (top or bottom" based on the `caption_valign` arg. This would eliminate the need for this captionvars list all together (as the only variable that would change based on the two settings is now `caption_valign`).